cypress/support/step_definitions/validations.md

```diff
@@ -445,6 +445,88 @@ While validating, if a key is present in both fcs-validation jsons (eg: cypress/
+
+Dynamic data :
+For dynamic data, the source of truth is fetched from the configModule's override function called "fetchDeviceDetails" explained [here](https://github.com/rdkcentral/firebolt-certification-suite?tab=readme-ov-file#request-overrides).
+
+### Usage :
+
+To fetch dynamic data from configModule, set an environment variable fetch_device_details_dynamically as true in the [supportConfig.json](../../../supportConfig.json). If this environment variable is true, dynamic details will be fetched from configModule based on the platform and override the existing environment variable deviceData, which holds the static data from <deviceMac.json>.
+
+If the dynamic device details fetching fails in configModule, the data will default to the static data. FCS expects the source of truth to be generated for the modules listed in [DYNAMIC_DEVICE_DETAILS_MODULES](../constants/constants.js). 
+
+
+### Implementation :
+
+The configModule should have an override function "fetchDeviceDetails" as explained [here](https://github.com/rdkcentral/firebolt-certification-suite?tab=readme-ov-file#request-overrides). This function should fetch the data and return it in the format specified below. If deviceId, deviceType, and distributor values need to be extracted, they should be saved as "DEVICEID", "DEVICE_TYPE", and "DEVICE_DISTRIBUTOR" keys, as these are the key names used in the validationObject. These values will then replace the values in <device-mac.json> stored in the environment variable "DEVICE_DATA" in FCS.
+
+When the test starts, deviceId is passed from FCS to this override function for the modules listed in [DYNAMIC_DEVICE_DETAILS_MODULES](../constants/constants.js) as the only parameter, which is later used to extract device details. If the data contains multiple objects, the required device details are fetched by mapping them with the corresponding deviceId passed from FCS to configModule.
+For example, if activeDevicedata is the dynamic device response, the data extraction will be as follows:
+
+#### Format
+
+```
+    if (activeDevicedata.id == deviceId) {
+      extractedData.DEVICEID = activeDevicedata.id;
+      extractedData.DEVICE_TYPE = activeDevicedata.data.deviceType;
+      extractedData.DEVICE_DISTRIBUTOR = deviceData[0].data.partner;
+
+    }
+    Cypress.env(CONSTANTS.DEVICE_DATA, extractedData);
+```
+
+### Example :
+From FCS, the override function is invoked in the following format:
+Format:
+  ```
+   {
+    "method": "fcs.fetchDeviceDetails",
+    "params": <deviceId>
+  }
+
+The response by overriding the static values in the environment will be as follows:
+Example:
+
+  ```
+  {
+    "DEVICEID": "354444327",
+    "DEVICE_TYPE": "ipstb",
+    "DEVICE_MODEL": "VALUE",
+     ...
```

Comment:
Use case : User want to add DEVICE_DISTRIBUTOR value to dynamic source of truth fetching. How can he add that ? Where is the explanation for that starting from writing a validation object with type : "DEVICE_DISTRIBUTOR" to returning the device_data containing the key DEVICE_DISTRIBUTOR ?

Reply:
create a new reamde file and provide link from here

Reply:
updated